src/layout/layout.rs

```diff
@@ -1482,31 +1486,131 @@ mod tests {
             #[case(Rect::new(0, 0, 2, 1), &[TEN, HALF], "bb")]
             #[case(Rect::new(0, 0, 2, 1), &[TEN, FULL], "bb")]
             #[case(Rect::new(0, 0, 2, 1), &[TEN, DOUBLE], "bb")]
-            #[case(Rect::new(0, 0, 2, 1), &[QUARTER, ZERO], "ab")]
-            #[case(Rect::new(0, 0, 2, 1), &[QUARTER, QUARTER], "ab")]
-            #[case(Rect::new(0, 0, 2, 1), &[QUARTER, HALF], "ab")]
-            #[case(Rect::new(0, 0, 2, 1), &[QUARTER, FULL], "ab")]
-            #[case(Rect::new(0, 0, 2, 1), &[QUARTER, DOUBLE], "ab")]
-            #[case(Rect::new(0, 0, 2, 1), &[THIRD, ZERO], "ab")]
-            #[case(Rect::new(0, 0, 2, 1), &[THIRD, QUARTER], "ab")]
-            #[case(Rect::new(0, 0, 2, 1), &[THIRD, HALF], "ab")]
-            #[case(Rect::new(0, 0, 2, 1), &[THIRD, FULL], "ab")]
-            #[case(Rect::new(0, 0, 2, 1), &[THIRD, DOUBLE], "ab")]
+            #[case(Rect::new(0, 0, 2, 1), &[QUARTER, ZERO], "bb")]
+            #[case(Rect::new(0, 0, 2, 1), &[QUARTER, QUARTER], "bb")]
+            #[case(Rect::new(0, 0, 2, 1), &[QUARTER, HALF], "bb")]
+            #[case(Rect::new(0, 0, 2, 1), &[QUARTER, FULL], "bb")]
+            #[case(Rect::new(0, 0, 2, 1), &[QUARTER, DOUBLE], "bb")]
+            #[case(Rect::new(0, 0, 2, 1), &[THIRD, ZERO], "bb")]
+            #[case(Rect::new(0, 0, 2, 1), &[THIRD, QUARTER], "bb")]
+            #[case(Rect::new(0, 0, 2, 1), &[THIRD, HALF], "bb")]
+            #[case(Rect::new(0, 0, 2, 1), &[THIRD, FULL], "bb")]
+            #[case(Rect::new(0, 0, 2, 1), &[THIRD, DOUBLE], "bb")]
```

Comment:
- 25/0 is wrong should be "aa"
- 25/25 is wrong should be "ab"
- 25/50 is wrong should be "ab"
- ...
- 33/0 is wrong
didn't look at all of these, but there's enough wrong rounding here to make this a bad result.
Not sure if it's the weight change or the 100x change that causes this - perhaps split the PR in two and make that more obvious.

As a pre-refactor on this it might be worth changing the constants to P000, P050, P100 etc. to make this more obviously broken. It could be worth also changing these to take constraints, width, result, description as params.

Reply:
These are all changes because of the rounding, not weights. I'll split the weights into a separate PR to make it clear.

I don't think we should pay too much attention to these tests unfortunately. All of these are over or under constrained tests, in a 2 pixel area. Even being 1 pixel off because of rounding gives results that look "wrong".

Reply:
If I use this following approach these tests don't break:

```rust
let start = ((start * 100.0).round() / 100.0).round() as u16;
let end = ((end * 100.0).round() / 100.0).round() as u16;
```

Reply:
Here's a commit with no tests changed but all tests passing with the weights and area_size changes reverted: https://github.com/ratatui-org/ratatui/pull/888/commits/fff20fef53192907304002ec0e2153683a33a2d1